src/pyodide/internal/python.js

```diff
@@ -82,6 +82,10 @@ export async function uploadArtifacts() {
   }
 }
 
+export function getMemorySnapshot() {
+  return new Uint8Array(MEMORY).slice();
```

Comment:
```suggestion
  return MEMORY;
```
Otherwise, this does two unnecessary copies:
1. we copied the heap when we assigned to it; we can do whatever we want with it now don't have to copy again
2. `MEMORY` is already a `Uint8Array` so `new Uint8Array(MEMORY)` copies it once and `slice()` copies it again.

There's even further copy reductions available now but I can do them in a different PR =)